opennms-model/src/main/java/org/opennms/netmgt/model/OnmsNode.java

```diff
@@ -323,6 +325,20 @@ public OnmsNode getParent() {
     public void setParent(OnmsNode parent) {
         m_parent = parent;
     }
+    @Transient
+    @XmlElement(name = "parent.id")
+    @JsonIgnore
+    public Integer getParentID() {
+
+        if( m_parent != null ) {
+           return m_parent.getId();
+        }
+        return null;
+    }
+
+    public void setParentID(Integer parentID) {
```

Comment:
Who is going to set this parentId ?

Can you add a test case in NodeRestServiceV2IT  to show that when ParentId is available, we can see it Rest endpoint.

Reply:
this parentID logic is changed ,we are getting from parent.getId() and setting it.
    public void setParentID(Integer parentID) {
        if( m_parent != null ) {
            this.m_parentID= m_parent.getId();
        }
    }

Reply:
![Screenshot from 2024-10-07 20-22-02](https://github.com/user-attachments/assets/89280283-0a7a-41c2-9c2c-fb71e89ca57e)

Reply:
i don't see any need for m_parentId and `setParentId`, we are only getting parentId from Rest API.

Also screenshot is not a replacement for test case.

Reply:
Yes ,it is correct removed unnecessary parameters.Will implement test case .it is just retrieval /v2/nodes for that we have to R&D to create node with parent .